### ISSUE TITLE
Allow overriding ContextName for default metrics

### ISSUE THREAD

**Issue by TDabasinskas** [AppMetrics/AspNetCore#34]
Looking into the source code, it seems that `Application` prefix for context names [is hardcoded](
https://github.com/AppMetrics/AspNetCore/blob/master/src/App.Metrics.AspNetCore.Core/Internal/HttpRequestMetricsRegistry.cs#L20):

```chsharp
 public static string ContextName = "Application.HttpRequests";
```

[Prometheus provides the guidelines for naming the metrics](https://prometheus.io/docs/practices/naming/) and one of the things these guidelines states is the following:

> A metric name. should have a (single-word) application prefix relevant to the domain the metric belongs to. The prefix is sometimes referred to as namespace by client libraries. For metrics specific to an application, the prefix is usually the application name itself. Sometimes, however, metrics are more generic, like standardized metrics exported by client libraries. Examples:
    - prometheus_notifications_total (specific to the Prometheus server)
    - process_cpu_seconds_total (exported by many client libraries)
    - http_request_duration_seconds (for all HTTP requests)

The hardcoded `Application` prefix for the metrics makes this somehow incompliant with the guidelines.

Would it be possible to support overriding of this, similarly it is possible to do with the custom metrics via `options.DefaultContextLabel`?

**Comment by alhardy** [comment]
Yeah guess we can allow the context names used by the asp.net core middleware to be overridden through configuration
